sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplication.java

```diff
@@ -3,16 +3,53 @@
 
 package com.azure.spring.sample.eventhubs.binder;
 
+import com.azure.spring.integration.core.api.reactor.Checkpointer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.messaging.Message;
+
+import java.util.function.Consumer;
+
+import static com.azure.spring.integration.core.AzureHeaders.CHECKPOINTER;
 
 /**
  * @author Warren Zhu
  */
 @SpringBootApplication
 public class EventHubBinderApplication {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventHubBinderApplication.class);
+
     public static void main(String[] args) {
         SpringApplication.run(EventHubBinderApplication.class, args);
     }
+
+    @Bean
+    public Consumer<Message<String>> consume() {
+        return message -> {
+            Checkpointer checkpointer = (Checkpointer) message.getHeaders().get(CHECKPOINTER);
+            LOGGER.info("New message received: '{}'", message);
+            checkpointer.success()
+                        .doOnSuccess(success -> LOGGER.info("Message '{}' successfully checkpointed", message))
+                        .doOnError(error -> LOGGER.error("Exception: {}", error.getMessage()))
+                        .subscribe();
+        };
+    }
+
+    // Replace destination with spring.cloud.stream.bindings.input.destination
+    // Replace group with spring.cloud.stream.bindings.input.group
+    @ServiceActivator(inputChannel = "{destination}.{group}.errors")
```

Comment:
Is the `@ServiceActivator` still the recommended way to implement error channel?